solidity/contracts/token/TokenBridgeCctp.sol

```diff
@@ -152,37 +163,46 @@ contract TokenBridgeCctp is HypERC20Collateral, AbstractCcipReadIsm {
             (bytes, bytes)
         );
 
-        bytes calldata tokenMessage = _hyperlaneMessage.body();
-        _validateMessageLength(tokenMessage);
-
         bytes29 originalMsg = TypedMemView.ref(cctpMessage, 0);
 
-        bytes29 burnMessage = originalMsg._messageBody();
-        require(
-            TokenMessage.amount(tokenMessage) == burnMessage._getAmount(),
-            "Invalid amount"
-        );
-        require(
-            TokenMessage.recipient(tokenMessage) ==
-                burnMessage._getMintRecipient(),
-            "Invalid recipient"
-        );
-
-        bytes32 sourceSender = burnMessage._getMessageSender();
-        require(sourceSender == _hyperlaneMessage.sender(), "Invalid sender");
-
+        uint32 origin = _hyperlaneMessage.origin();
         uint32 sourceDomain = originalMsg._sourceDomain();
         require(
-            sourceDomain ==
-                hyperlaneDomainToCircleDomain(_hyperlaneMessage.origin()),
+            sourceDomain == hyperlaneDomainToCircleDomain(origin),
             "Invalid source domain"
         );
 
         uint64 sourceNonce = originalMsg._nonce();
-        require(
-            sourceNonce == uint64(bytes8(TokenMessage.metadata(tokenMessage))),
-            "Invalid nonce"
-        );
+
+        address circleRecipient = originalMsg._recipient().bytes32ToAddress();
+        if (circleRecipient == address(tokenMessenger)) {
```

Comment:
Imo not obvious how this logic works, so would like to see more comments on the semantics of the different messages